File: src/main/java/com/fitnessdump/FitnessDump/Service/CalorieCalculatorService.java
package com.fitnessdump.FitnessDump.Service;

import com.fitnessdump.FitnessDump.DTOs.Meals.MealPlanDTO;
import com.fitnessdump.FitnessDump.DTOs.Nutrition.CalorieCalculatorRequestDTO;
import com.fitnessdump.FitnessDump.DTOs.Nutrition.CalorieCalculatorResponseDTO;
import com.fitnessdump.FitnessDump.DTOs.Nutrition.MealPlanGenerationConfigDTO;
import com.fitnessdump.FitnessDump.DTOs.Training.TrainingProgramDTO;

public interface CalorieCalculatorService {

    CalorieCalculatorResponseDTO calculateNutrition(CalorieCalculatorRequestDTO request);

    CalorieCalculatorResponseDTO calculateAndSaveToPersonalSettings(Long userId, CalorieCalculatorRequestDTO request);

    MealPlanDTO generateMealPlan(Long userId, CalorieCalculatorRequestDTO request);

    TrainingProgramDTO generateTrainingProgram(Long userId, CalorieCalculatorRequestDTO request);

    void generateDailyPlan(Long userId, CalorieCalculatorRequestDTO request);

    MealPlanDTO generateSmartMealPlan(Long userId, CalorieCalculatorRequestDTO request, boolean includeWorkoutDays);

    // New configurable methods
    MealPlanDTO generateMealPlanWithConfig(Long userId, CalorieCalculatorRequestDTO request,
            MealPlanGenerationConfigDTO config);

    MealPlanDTO generateSmartMealPlanWithConfig(Long userId, CalorieCalculatorRequestDTO request,
            MealPlanGenerationConfigDTO config);
}

File: frontend/src/services/calorie-calculator.service.ts
import axios from '../config/axios.config';
import { API_CONFIG } from '../config/api.config';
import { CalorieCalculatorRequestDTO, CalorieCalculatorResponseDTO } from '../types/personal-settings.types';
import { MealPlanGenerationConfigDTO } from '../types/program.types';

const calculateNutrition = (data: CalorieCalculatorRequestDTO) =>
  axios.post<CalorieCalculatorResponseDTO>(`${API_CONFIG.BASE_URL}/calorie-calculator/calculate`, data).then(r => r.data);

const calculateAndSaveToPersonalSettings = (userId: number, data: CalorieCalculatorRequestDTO) =>
  axios.post<CalorieCalculatorResponseDTO>(`${API_CONFIG.BASE_URL}/calorie-calculator/calculate-and-save/${userId}`, data).then(r => r.data);

const generateTrainingProgram = (userId: number, data: CalorieCalculatorRequestDTO) =>
  axios.post(`${API_CONFIG.BASE_URL}/calorie-calculator/generate-training-program/${userId}`, data).then(r => r.data);

const generateMealPlan = (userId: number, data: CalorieCalculatorRequestDTO) =>
  axios.post(`${API_CONFIG.BASE_URL}/calorie-calculator/generate-meal-plan/${userId}`, data).then(r => r.data);

const generateSmartMealPlan = (userId: number, data: CalorieCalculatorRequestDTO, includeWorkoutDays: boolean = true) =>
  axios.post(`${API_CONFIG.BASE_URL}/calorie-calculator/generate-smart-meal-plan/${userId}`, data, {
    params: { includeWorkoutDays }
  }).then(r => r.data);

const generateSmartMealPlanWithConfig = (userId: number, request: CalorieCalculatorRequestDTO, config: MealPlanGenerationConfigDTO) =>
  axios.post(`${API_CONFIG.BASE_URL}/calorie-calculator/generate-smart-meal-plan-with-config/${userId}`, {
    request,
    config
  }).then(r => r.data);

const generateMealPlanWithConfig = (userId: number, request: CalorieCalculatorRequestDTO, config: MealPlanGenerationConfigDTO) =>
  axios.post(`${API_CONFIG.BASE_URL}/calorie-calculator/generate-meal-plan-with-config/${userId}`, {
    request,
    config
  }).then(r => r.data);

const generateDailyPlan = (userId: number, data: CalorieCalculatorRequestDTO) =>
  axios.post(`${API_CONFIG.BASE_URL}/calorie-calculator/generate-daily-plan/${userId}`, data).then(r => r.data);

const calorieCalculatorService = {
  calculateNutrition,
  calculateAndSaveToPersonalSettings,
  generateTrainingProgram,
  generateMealPlan,
  generateSmartMealPlan,
  generateSmartMealPlanWithConfig,
  generateMealPlanWithConfig,
  generateDailyPlan
};

export default calorieCalculatorService; 